### DISCUSSION TITLE
AssemblyAttribute automatically added to AssemblyInfo of a non-Module project

### DISCUSSION THREAD

**Discussion by giantpenguin** [OrchardCMS/OrchardCore#14443] [Q&A]
The OrchardCore.Module.Targets.targets has its Target OrchardCoreEmbedModuleAssets, this target has a ItemGroup to help automatically generate 2 assembly lines to the AssemblyInfo.cs like 
    **[assembly: OrchardCore.Modules.Manifest.ModuleMarkerAttribute("...") ]
    [assembly: OrchardCore.Modules.Manifest.ModuleAssetAttribute("...") ]** 
this should be intended only to every Module (has Manifest.cs file in its project).

But, this is a circumstance: 
- the Lombiq.TrainingDemo is a Module.
- the Lombiq.TrainingDemo depends on the Lombiq.HelpfulLibraries.OrchardCore.
- the Lombiq.HelpfulLibraries.OrchardCore depends on Lombiq.HelpfulLibraries.AspNetCore, Lombiq.HelpfulLibraries.Common.
- the Lombiq.OSOCE.Web is a website project to test the Lombiq.TrainingDemo Module.
- the very first build of the Lombiq.OSOCE.Web project, the 2lines of above  **[assembly ...]** are automatically added to the generated AssemblyInfo.cs files of the Lombiq.HelpfulLibraries.AspNetCore, Lombiq.HelpfulLibraries.Common projects => this is a wrong because these 2projects have no dependent/reference on OrchardCore then cause the build fail, the error is in the generated AssemblyInfo.cs (under the obj folder).

My question is: **how to condition that OrchardCore.Module.Targets.targets to only add the [assembly ...] lines to Module project (which has dependent/reference on OrchardCore)?
Should I bring this as a bug to OrchardCore team?**

**Comment by Piedone** [comment]
Hmm, what exactly is the build error you're seeing? But yeah, in this scenario, those marker attributes shouldn't be on the Helpf

To be clear, none of the mentioned Helpful Libraries projects reference `OrchardCore.Module.Targets` but `Lombiq.HelpfulLibraries.OrchardCore` does reference several OC packages, including modules ones ([see here](https://github.com/Lombiq/Helpful-Libraries/blob/dev/Lombiq.HelpfulLibraries.OrchardCore/Lombiq.HelpfulLibraries.OrchardCore.csproj#L27)). However, the problematic projects mentioned above don't have OC references at all (see [`Lombiq.HelpfulLibraries.AspNetCore`](https://github.com/Lombiq/Helpful-Libraries/blob/dev/Lombiq.HelpfulLibraries.AspNetCore/Lombiq.HelpfulLibraries.AspNetCore.csproj) and [`Lombiq.HelpfulLibraries.Common`](https://github.com/Lombiq/Helpful-Libraries/blob/dev/Lombiq.HelpfulLibraries.Common/Lombiq.HelpfulLibraries.Common.csproj)).

**Reply by giantpenguin** [reply]
Have a great new week @Piedone!

Yeah, so it is a sense that the marker attributes should be only added to OC Modules' AssemblyInfo but not to others.

This is an auto generated AssemblyInfo (in the Lombiq.HelpfulLibraries.Common project) which has been added the 2 marker attributes 

`//------------------------------------------------------------------------------
// <auto-generated>
//     This code was generated by a tool.
//     Runtime Version:4.0.30319.42000
//
//     Changes to this file may cause incorrect behavior and will be lost if
//     the code is regenerated.
// </auto-generated>
//------------------------------------------------------------------------------

using System;
using System.Reflection;

[assembly: OrchardCore.Modules.Manifest.ModuleMarkerAttribute("Lombiq.HelpfulLibraries.Common", "")]
[assembly: OrchardCore.Modules.Manifest.ModuleAssetAttribute("")]
[assembly: System.Reflection.AssemblyCompanyAttribute("Lombiq Technologies")]
[assembly: System.Reflection.AssemblyConfigurationAttribute("Debug")]
[assembly: System.Reflection.AssemblyCopyrightAttribute("Copyright © 2011, Lombiq Technologies Ltd.")]
[assembly: System.Reflection.AssemblyDescriptionAttribute("Lombiq Helpful Libraries - Common Libraries: Various useful libraries that can be" +
    " handy when developing for .NET. See the project website for detailed documentat" +
    "ion.")]
[assembly: System.Reflection.AssemblyFileVersionAttribute("1.0.0.0")]
[assembly: System.Reflection.AssemblyInformationalVersionAttribute("1.0.0")]
[assembly: System.Reflection.AssemblyProductAttribute("Lombiq.HelpfulLibraries.Common")]
[assembly: System.Reflection.AssemblyTitleAttribute("Lombiq.HelpfulLibraries.Common")]
[assembly: System.Reflection.AssemblyVersionAttribute("1.0.0.0")]
[assembly: System.Reflection.AssemblyMetadataAttribute("RepositoryUrl", "https://github.com/Lombiq/Helpful-Libraries")]

// Generated by the MSBuild WriteCodeFragment class.
`
which causes this error when building consumer project(s) e.g., Lombiq.HelpfulLibraries project

Severity	Code	Description	Project	File	Line	Suppression State
Error	CS0246	The type or namespace name 'OrchardCore' could not be found (are you missing a using directive or an assembly reference?)	Lombiq.HelpfulLibraries.Common	C:\ZRepos\OrchardEval\Lombiq\Lombiq.HelpfulLibraries.Common\obj\Debug\net6.0\Lombiq.HelpfulLibraries.Common.AssemblyInfo.cs	14	Active

**Reply by Piedone** [reply]
Yeah, that definitely shouldn't happen. However, it doesn't happen for me in [OSOCE](https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions). Can you share the full solution you're working with?

BTW if you don't need or want to work with the code, you can just add HL [from NuGet](https://www.nuget.org/packages/Lombiq.HelpfulLibraries.Common/).

Have a nice week :).

**Reply by giantpenguin** [reply]
I have my [Orchard_Eval](https://github.com/giantpenguin/Orchard_Eval) repository privately that I have invited your email [zoltan.lehoczky@lombiq.com](mailto:zoltan.lehoczky@lombiq.com), please accept and take a look.
The repo has all the OC projects of version 1.7, the Lombiq projects and a testing solution (LombiqEval_Src.sln) are in the Lombiq folder.

I chose way of projects references (not packages) for evaluating and made many changes to the .csproj, .props, .targets to have a test site work so I am not totally confident where this error come from. But just deleting the obj folder the issue will occur (because of a new AssemblyInfo generated).

Thanks & regards,

**Reply by Piedone** [reply]
I don't see the Lombiq projects added to the solution there. However, those projects load Orchard Core from NuGet. Mixing that together with the full Orchard Core source code will cause all kinds of mismatches. You can attempt to make it work by changing all the OC package references to project references. However, in your place I'd rather use OC from packages instead.

**Comment by giantpenguin** [comment]
Hi Zoltán,

I believe that the solution has been cleaned to use all the project
references of OC
![image](https://github.com/OrchardCMS/OrchardCore/assets/7245621/0094955f-d000-48ad-bee0-6dd8eca64617)
![image](https://github.com/OrchardCMS/OrchardCore/assets/7245621/5c04d914-16ac-428f-87c7-c143ba0b37fd)

This issue is just a little weird, I think we should let it aside for
bigger ones.

I appreciate your support!

Best regards,

**Reply by Piedone** [reply]
Sure :).

**Comment by giantpenguin** [comment]
The issue does happen when I mix library/helper projects (e.g., Lombiq.HelpfulLibraries.AspNetCore, Lombiq.HelpfulLibraries.Common) with the "Module" projects since the "Module" projects (in case using ProjectReference, not PackageReference) need import (through the .csproj file) the OrchardCore.Module.Targets.targets which has its a target to automatically update the AssemblyInfo.cs.

For solving the issue (this is like the Orchard Module projects are in the same OrchardCore.Modules folder): Have the "Module" projects separately in a folder and have Directory.Build.props Directory.Build.targets (which imports the OrchardCore.Module.Targets.targets) in this folder.
